### ISSUE TITLE
BasicBackwardsCompatibilityIT.testIndexRollingUpgrade sometimes fails

### ISSUE THREAD

**Issue by colings86** [elastic/elasticsearch#17130]
Example failure:
http://build-us-00.elastic.co/job/es_core_2x_window-2008/712

Reproduce command (I can't reproduce this locally):

```
REPRODUCE WITH: mvn verify -Pdev -Dskip.unit.tests -pl org.elasticsearch.qa.backwards:2.0 -Dtests.seed=FE8754AE850E4A36 -Dtests.class=org.elasticsearch.bwcompat.BasicBackwardsCompatibilityIT -Dtests.method="testIndexRollingUpgrade" -Des.logger.level=DEBUG -Dtests.nightly=false -Dtests.client.ratio=0.0 -Dtests.heap.size=512m -Des.node.mode=local -Dtests.jvm.argline="-server -XX:+UseG1GC -XX:+UseCompressedOops -Djava.net.preferIPv4Stack=true" -Dtests.locale=sr-BA -Dtests.timezone=Australia/Tasmania
```

Failure:

```
FAILURE 11.8s | BasicBackwardsCompatibilityIT.testIndexRollingUpgrade <<<
   > Throwable #1: java.lang.AssertionError: Count is 96 but 116 was expected.  Total shards: 14 Successful shards: 14 & 0 shard failures:
   > 	at __randomizedtesting.SeedInfo.seed([FE8754AE850E4A36:88FA0EA86CDC10E0]:0)
   > 	at org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount(ElasticsearchAssertions.java:244)
   > 	at org.elasticsearch.bwcompat.BasicBackwardsCompatibilityIT.testIndexRollingUpgrade(BasicBackwardsCompatibilityIT.java:371)
   > 	at java.lang.Thread.run(Thread.java:745)
```

**Comment by brwe** [comment]
I can reproduce it. Will try to figure it out.

**Comment by brwe** [comment]
This fails because of https://github.com/elastic/elasticsearch/pull/15832.  That fix was backported to 2.0 branch but we never made a release with this fix. So now we are still testing against 2.0.2 which does not have the fix. I do not remember why we did not think of that or why it starts failing only now and did not before. Maybe something in CI changed? Anyway, I will change the test to flush before upgrading if version is < 2.0.3 to make sure we don't run into it.
